lambdas/data-migration2/tests/test-granulesAndFiles.js

```diff
@@ -0,0 +1,538 @@
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 0,
+      failed: 0,
+      skipped: 0,
+      success: 0,
+    },
+    granulesSummary: {
+      dynamoRecords: 1,
+      failed: 0,
+      skipped: 1,
+      success: 0,
+    },
+  });
+
+  const records = await knex(tableNames.granules);
+  t.is(records.length, 1);
+});
+
+test.serial('migrateGranulesAndFiles processes multiple granules and files', async (t) => {
+  const {
+    collectionPgModel,
+    executionPgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const alternateBucket = cryptoRandomString({ length: 10 });
+  await s3Utils.createBucket(alternateBucket);
+
+  const testExecution2 = fakeExecutionRecordFactory();
+  await executionPgModel.create(
+    knex,
+    testExecution2
+  );
+
+  const testCollection2 = fakeCollectionRecordFactory();
+  await collectionPgModel.create(
+    t.context.knex,
+    testCollection2
+  );
+
+  const testGranule1 = testGranule;
+  const testGranule2 = generateTestGranule(testCollection2, testExecution2.arn, alternateBucket);
+  const files = testGranule1.files.concat(testGranule2.files);
+
+  await Promise.all(files.flatMap((file) => s3Utils.s3PutObject({
+    Bucket: file.bucket,
+    Key: file.key,
+    Body: 'some-body',
+  })));
+
+  await Promise.all([
+    granulesModel.create(testGranule1),
+    granulesModel.create(testGranule2),
+  ]);
+
+  t.teardown(async () => {
+    granulesModel.delete({ granuleId: testGranule1.granuleId });
+    granulesModel.delete({ granuleId: testGranule2.granuleId });
+    await s3Utils.recursivelyDeleteS3Bucket(alternateBucket);
+  });
+
+  const migrationSummary = await migrateGranulesAndFiles(process.env, knex);
+  t.deepEqual(migrationSummary, {
+    filesSummary: {
+      dynamoRecords: 2,
+      failed: 0,
+      skipped: 0,
+      success: 2,
+    },
+    granulesSummary: {
+      dynamoRecords: 2,
+      failed: 0,
+      skipped: 0,
+      success: 2,
+    },
+  });
+  const records = await knex(tableNames.granules);
+  const fileRecords = await knex(tableNames.files);
+  t.is(records.length, 2);
+  t.is(fileRecords.length, 2);
+});
+
+test.serial('migrateGranulesAndFiles processes all non-failing granule records and does not process files of failing granule records', async (t) => {
+  const {
+    collectionPgModel,
+    executionPgModel,
+    knex,
+    testGranule,
+  } = t.context;
+
+  const alternateBucket = cryptoRandomString({ length: 10 });
+  await s3Utils.createBucket(alternateBucket);
+
+  const testExecution2 = fakeExecutionRecordFactory();
+  await executionPgModel.create(
+    knex,
+    testExecution2
+  );
+
+  const testCollection2 = fakeCollectionRecordFactory();
+  await collectionPgModel.create(
+    t.context.knex,
+    testCollection2
+  );
+
+  const testGranule2 = generateTestGranule(testCollection2, testExecution2.arn, alternateBucket);
```

Comment:
I don't think you really need a new collection, execution, or bucket for the conditions of this test to hold? You just need one granule to fail and the other to succeed

Reply:
👀 Ah there I go again creating unnecessary things for my test [`5374077` (#2051)](https://github.com/nasa/cumulus/pull/2051/commits/53740775e8529f9099375c223a3b91bb42e27b0e)